proto/spire/plugin/server/nodeattestor/v1/nodeattestor.proto

```diff
@@ -1,42 +1,54 @@
-/** Responsible for validating the Node Agent’s Attested Data. */
-
 syntax = "proto3";
-package spire.agent.nodeattestor;
-option go_package = "github.com/spiffe/spire/proto/spire/server/nodeattestor";
+package spire.plugin.server.nodeattestor.v1;
+option go_package = "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/server/nodeattestor/v1;nodeattestorv1";
 
-import "spire/common/plugin/plugin.proto";
-import "spire/common/common.proto";
+service NodeAttestor {
+    // Attest attests attestation payload received from the agent and
+    // optionally participates in challenge/response attestation mechanics.
+    //
+    // The attestation flow is as follows:
+    // 1. SPIRE server opens up a stream to the plugin via Attest.
+    // 2. SPIRE server sends a request containing the attestation payload
+    //    received from the agent.
+    // 3. Optionally, the plugin responds with a challenge:
+    //    3a. SPIRE server sends the challenge to the agent.
+    //    3b. The agent responds with the challenge response.
+    //    3c. SPIRE server sends the challenge response to the plugin.
+    //    3d. Step 3 is repeated until the plugin is satisfied and does
+    //        not respond with an additional challenge.
+    // 4. The plugin returns the attestation results to SPIRE server and closes
+    //    the stream.
+    rpc Attest(stream AttestRequest) returns (stream AttestResponse);
+}
 
-/** Represents a request to attest a node. */
 message AttestRequest {
-    reserved 2;
-
-    /** A type which contains attestation data for specific platform. */
-    spire.common.AttestationData attestation_data = 1;
-    /** Challenge response */
-    bytes response = 3;
+    oneof request {
+        // Required in the first request. The attestation payload. See the
+        // Attest RPC for details.
+        bytes payload = 1;
+
+        // Required in subsequent requests. The response to a plugin issued
+        // challenge. See the Attest RPC for details.
+        bytes response = 2;
+    }
 }
 
-/** Represents a response when attesting a node.*/
 message AttestResponse {
-    reserved 1;
-
-    /** SPIFFE ID of the attested node */
-    string agent_id = 2;
-
-    /** Challenge required for attestation */
-    bytes challenge = 3;
-
-    /** Optional list of selectors */
-    repeated spire.common.Selector selectors = 4;
+    oneof response {
+        // Required in all but the last response. The challenge to issue the
+        // agent. See the Attest RPC for details.
+        bytes challenge = 2;
```

Comment:
Is the ordering intentional?

Reply:
It was... I wanted the order to help convey that, when present, challenges come before the result. Though if I keep it I should probably swap the field values.

What are your thoughts on the order?